### ISSUE TITLE
[BUG] - Modal scrollBehavior="outside" breaks placement prop

### ISSUE THREAD

**Issue by robodove** [nextui-org/nextui#1548]
### NextUI Version

latest

### Describe the bug

By using the scrollBehavior="outside" prop, placement is always at the top (with a my-16 margin)

Cause:
https://github.com/nextui-org/nextui/blob/main/packages/core/theme/src/components/modal.ts#L171

For now the only way to fix it on the user side is to not use scrollBehavior="outside" but enter 
```tsx
<Modal classNames={{
 wrapper: "items-end overflow-y-auto",
      base: "my-1",
}} />
```
for example if one prefers placement="bottom" and scrollBehavior="outside"

### Steps to Reproduce the Bug or Issue

1. Create a <Modal with placement="bottom" />
2. Add scrollBehaviour="bottom"
3. Modal will be placed at top

### Expected behavior

Modal should always respect the placement prop

### Operating System Version

any

### Browser

Other (add additonal context)

**Comment by CHENFANGC** [comment]
It might be due to incorrect usage of parameters. `bottom` is not a valid parameter for `scrollBehavior`. The valid parameters for `scrollBehavior `are `normal` | `inside` | `outside`
https://nextui.org/docs/components/modal#api
![image](https://github.com/nextui-org/nextui/assets/29766018/9330ca15-2096-4c77-bf44-a15a563b3e0c)

**Comment by robodove** [comment]
> It might be due to incorrect usage of parameters. `bottom` is not a valid parameter for `scrollBehavior`. The valid parameters for `scrollBehavior `are `normal` | `inside` | `outside`
> 
> https://nextui.org/docs/components/modal#api
> 
> ![image](https://github.com/nextui-org/nextui/assets/29766018/9330ca15-2096-4c77-bf44-a15a563b3e0c)
> 
> 

No. The bug is emerging when scrollBehavior=outside and placement=bottom is used together.

Look at your screenshot of the docs page, the prop just below scrollBehavior

**Comment by CHENFANGC** [comment]
> scrollBehavior=“outside”

Thank you for correcting the reproduction steps. 
https://codesandbox.io/p/sandbox/sleepy-hugle-d6tt88?file=/App.jsx:37,42&utm_medium=sandpack
There are two scenarios:

**Scenario 1**: When the text does not exceed the container. In this case, `scrollBehavior='outside'` should not be used.
placement='top'
![image](https://github.com/nextui-org/nextui/assets/29766018/7bf31dc6-6a55-4a95-9777-39e32414a0be)

placement='bottom'
![image](https://github.com/nextui-org/nextui/assets/29766018/36696c27-baf6-4a1c-a1fb-e008ee82cda7)

In this case, the `placement` property is effective.

**Scenario 2**: When the text exceeds the container. In this case, the `scrollBehavior` property must be added. When `scrollBehavior='outside'` and `placement='bottom'` are used together, the behavior is such that the `placement` property does not take effect. Regardless of the `placement` value set, the behavior remains the same, with a distance from the top, and a scroll-down state, as shown in the image below.
![image](https://github.com/nextui-org/nextui/assets/29766018/f36e85c6-3c48-4c8f-806a-44a0ad2c3047)
![image](https://github.com/nextui-org/nextui/assets/29766018/5b411879-d010-48df-a2a7-9f03d969f205)

`scrollBehavior='outside'` and `placement='top'`
![image](https://github.com/nextui-org/nextui/assets/29766018/a82e788e-6fa2-4a5c-9bfb-32ee3fc3086e)
![image](https://github.com/nextui-org/nextui/assets/29766018/0f953c83-cfea-428f-aeb4-e33ac20a1ace)

In this case, the `placement` property does not take effect.

Is it your expectation that when setting `scrollBehavior='outside'` and `placement='bottom'`, the initial pop-up state is at the bottom of the scrollbar, as shown in the image below?
![image](https://github.com/nextui-org/nextui/assets/29766018/e7948ef6-c328-4665-842f-897afc7ce76c)

**Comment by abhinav700** [comment]
@wingkwong Can I work on this issue and add the 3 cases suggested by @robodove ?

**Comment by abhinav700** [comment]
@wingkwong This behavior is taking place because of the `items-start` class in `scrollBehavior:outside` variant and disabling it should provide the solution for OP.

Should we allow the user to change the placement property when `scrollBehavior = outside`, or we should keep the things unchanged?
![image](https://github.com/user-attachments/assets/7e77b694-43b0-498c-a75d-845857abe34a)

**Comment by tordsta** [comment]
Any updates on this bug?

**Comment by sergiozeppo** [comment]
The bug’s still there, it just showed up again. 
BTW, thanks @alightinastorm for the tip on how to avoid that.

**Comment by pablojsx** [comment]
I confirm, the bug is still there although i'm using v2.7.5 can't upgrade to tailwind v4 now

**Comment by wingkwong** [comment]
Closing in favor of v3. If the issue is reproducible in v3, please create a new one with a minimal reproducible example.
